### DISCUSSION TITLE
How to tag a golang struct field with CUE expression referencing another field

### DISCUSSION THREAD

**Discussion by sjwl** [cue-lang/cue#2188] [Q&A]
I want a golang app that when calling `cue get go --local` generates the following CUE config

```cue
package foo

#Foo: {
  name: string
  altName: string | *name
}
```

The closest I can get is this:

`main.go`
```
package foo

type Foo struct {
	Name    string `cue:"" json:"name"`
	AltName string `cue:"string | *name" json:"altName"`
}
```

which generates
`main_go_gen.cue`
```
// Code generated by cue get go. DO NOT EDIT.

//cue:generate cue get go example.com/foo

package foo

#Foo: {
	"name":  string                    @go(Name)
	altName: string & (string | *name) @go(AltName)
}
```

which unfortunately gives CUE error
```
% cue eval main_go_gen.cue
#Foo.altName: reference "name" not found:
    ./main_go_gen.cue:9:31
```

**Comment by davildaran** [comment]
I have the same/similar question [posted here](https://github.com/cue-lang/cue/issues/2740)

I've tried to reference in the original Go module like `#Foo.name` in the cue annotation. The generated cue schema no longer generates the `"name"` syntax for that struct field, however, on `cue vet ...` and presumably other commands it will not work as expected
